api-extractor/src/definitions/ApiItem.ts

```diff
@@ -237,6 +264,41 @@ abstract class ApiItem {
   protected reportWarning(message: string): void {
     this.warnings.push(message);
   }
+
+  /**
+   * This function assumes all references from this ApiItem have been resolved and we can now safely create 
+   * the documentation.
+   */
+  @virtual
+  protected onResolveReferences(): void {
+    this.documentation = new ApiDocumentation(this, this.extractor.docItemLoader, this.extractor, this.reportError);
+    // this.collectTypeReferences(this); 
+  }
+
+  /**
+   * Some ApiItems refer to the documentation of other ApiItems that are within the same package. To do that, 
+   * we have to make sure that the documentation is resolved for the reffered to ApiItem before we can 
+   * refer to it. This function makes sure we create the documentation for each ApiItem in the correct order.
+   * In the event that a circular dependency occurs, an error is reported. For example, if ApiItemOne has 
+   * an \@inheritdoc referencing ApiItemTwo, and ApiItemTwo has an \@inheritdoc refercing ApiItemOne then 
+   * we have a circular dependency and an error will be reported.
+   */
+  public resolveReferences(): void {
+    switch (this._state) {
+      case ResolveState.Resolved:
+        return;
+      case ResolveState.Unresolved:
+        this._state = ResolveState.Resolving;
+        this.onResolveReferences();
+        this._state = ResolveState.Resolved;
+        return;
+      case ResolveState.Resolving:
+        this.reportError('@inheritdoc has a circular reference');
```

Comment:
\@inheritdoc isn't the only thing that can call resolveReferences(); this error could be misleading.

It also might be useful to create a couple errors and look at the line numbers and error message, to make sure it makes sense.